charts/baseChart.js

```diff
@@ -0,0 +1,56 @@
+import 'highcharts';
+import React from 'react';
+import ReactHighcharts from 'react-highcharts';
+
+/**
+ * Component representing a single chart
+ */
+module.exports = class Chart extends React.Component {
+    /**
+     * Create a new chart
+     * @param props props of the chart
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: null,
+            error: null
+        };
+    }
+
+    /**
+     * @inheritdoc
+     */
+    render() {
+        if (this.state.error) {
+            return (
+                <div>Error: {this.state.error}</div>
+            );
+        }
+        if (this.state.data) {
+            return (
+                <ReactHighcharts config =
+                    {this.getChartConfig()}></ReactHighcharts>
+            );
+        }
+
+        return (
+            <div>Loading...</div>
+        );
+    }
+
+    /**
+     * Retrieve results from the given endpoint.  If retrieved successfully,
+     * transform the response to json and apply the given function on the
+     * results.  Returns a promise with the results.
+     */
+    _retrieveResultsHelper(endpoint, jsonMapFunc) {
+        return fetch(endpoint)
+            .then(res => {
+                if (res.status === 200) {
+                    return res.json()
+                        .then(jsonData => jsonData.map(jsonMapFunc));
+                }
+            });
+    }
+};
```

Comment:
If you're going to use different casing for filenames, I suggest using TitleCase for files that export a constructor.

Reply:
not sure what you mean here, filenames are consistently camelCase everywhere.  are you saying TitleCase for ones exporting a constructor and camelCase for others?

Reply:
If camelCase is the convention you want as a blanket rule then it's fine, although it would contract with jitsi-meet.